pkg/execution/state/driver_response.go

```diff
@@ -381,6 +401,24 @@ func (r *DriverResponse) Final() bool {
 	return false
 }
 
+// SingleStep returns a single generator op if this response is a generator
+// containing only one op, otherwise nil.
+func (r *DriverResponse) SingleStep() *GeneratorOpcode {
+	if r.Generator == nil || len(r.Generator) != 1 {
+		return nil
+	}
+	return r.Generator[0]
+}
+
+// IsSingleStepError returns whether this response is an error from running a
+// step.
+func (r *DriverResponse) IsSingleStepError() bool {
+	if step := r.SingleStep(); step != nil && isJsonNonNullish(step.Error) {
```

Comment:
Why do we have to call` isJsonNonNullish` every time?  Shouldn't we be able to track whether this is a step error by looking at the opcode?